### PR TITLE
Expand Audit Log features

### PR DESCRIPTION
**Pull request by dcramer**
- Move to React
- Pagination
- Event type filter

/cc @getsentry/ui @getsentry/api

screenshot 2016-04-15 11 16 25

### CODE REVIEW

**Comment by dcramer** [comment]
TODO:

- Switch to @macqueen's new avatar stuff (once it lands)
- Add a default avatar here for non-actor based stuff

File: src/sentry/static/sentry/app/views/organizationAuditLog.jsx [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+  },
+
+  componentWillMount() {
+    this.fetchData();
+  },
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.location.search !== this.props.location.search ||
+        nextProps.params.orgId !== this.props.params.orgId) {
+      this.remountComponent();
```
Comment (dcramer) [inline src/sentry/static/sentry/app/views/organizationAuditLog.jsx]: @benvinegar will probably be mad about this, but I stand by the simplicity of it :)

File: src/sentry/static/sentry/less/includes/select2.less [hunk 2]
```diff
@@ -11,7 +11,6 @@
   padding: 4px 10px 3px;
   top: -1px;
   box-shadow: 0 1px 1px rgba(0, 0, 0, .05);
-  max-width: 180px;
```
Comment (dcramer) [inline src/sentry/static/sentry/less/includes/select2.less]: @ckj fyi this changes some pages -- full width select2 isn't ideal, but we can fix them by just setting a width (vs limiting what select2 can do in general). I didn't think it was bad enough to go change them.
screenshot 2016-04-15 11 02 16

**Comment by dcramer** [comment]
k all set here

**Comment by mattrobenolt** [comment]
:+1:

File: src/sentry/static/sentry/app/views/organizationAuditLog.jsx [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+      <DocumentTitle title={this.getTitle()}>
+        <OrganizationHomeContainer>
+          <h3>{t('Audit Log')}</h3>
+
+          <div className="pull-right">
+            <form className="form-horizontal" style={{marginBottom: 20}}>
+              <div className="control-group">
+                <div className="controls">
+                  <SelectInput name="event" onChange={this.onEventSelect}
+                               value={currentEventType} style={{width: 250}}>
```
Comment (mitsuhiko) [inline src/sentry/static/sentry/app/views/organizationAuditLog.jsx]: are these not supposed to be "250px"?

File: src/sentry/static/sentry/app/components/selectInput.jsx [hunk 4]
```diff
@@ -61,17 +63,8 @@ const SelectInput = React.createClass({
...
-        disabled: this.props.disabled,
-        required: this.props.required,
-        multiple: this.props.multiple,
-        placeholder: this.props.placeholder,
-        className: this.props.className,
-        value: this.props.value,
-    };
     return (
-      <select {...opts}>
+      <select ref="select" {...this.props}>
```
Comment (mitsuhiko) [inline src/sentry/static/sentry/app/components/selectInput.jsx]: afaik `...this.props` forwards children too.

Comment (dcramer) [inline src/sentry/static/sentry/app/components/selectInput.jsx]: idk if you can do children={} to bind it though. This was just convenience so we stopped missing props

File: src/sentry/static/sentry/app/views/organizationAuditLog.jsx (hunk 3 above)
Comment (dcramer) [inline src/sentry/static/sentry/app/views/organizationAuditLog.jsx]: nope it maps to JS so "250" translates to "250px"

**Pull request merged**